client/src/components/advancedSelectWidget/AdvancedSelect.js

```diff
@@ -163,180 +145,146 @@ export default class AdvancedSelect extends Component {
-                        renderRow={overlayRenderRow}
-                        isLoading={isLoading}
-                        loaderMessage="No results found"
-                        tableClassName={overlayTableClassName}
-                      />
-                      {this.paginationFor(filterType)}
-                    </Col>
-                  </Row>
-                )
+    return (
+      <>
+        <div id={`${fieldName}-popover`}>
+          <InputGroup>
+            <Popover
+              className="advanced-select-popover"
+              popoverClassName="bp3-popover-content-sizing"
+              content={advancedSearchPopoverContent}
+              isOpen={showOverlay}
+              captureDismiss
+              interactionKind={PopoverInteractionKind.CLICK}
+              onInteraction={this.handleInteraction}
+              usePortal={false}
+              position={Position.BOTTOM}
+              modifiers={{
+                preventOverflow: {
+                  enabled: false
+                },
+                flip: {
+                  enabled: false
+                }
               }}
-            </ContainerDimensions>
-          </Popover>
-        </Overlay>
+            >
+              <FormControl
+                name={fieldName}
+                value={searchTerms || ""}
+                placeholder={placeholder}
+                onChange={this.changeSearchTerms}
+                onFocus={this.handleInputFocus}
+                inputRef={this.searchInput}
+              />
+            </Popover>
+            {extraAddon && <InputGroup.Addon>{extraAddon}</InputGroup.Addon>}
+            {addon && (
+              <FieldHelper.FieldAddon fieldId={fieldName} addon={addon} />
+            )}
+          </InputGroup>
+        </div>
         <AdvancedSelectTarget overlayRef={this.overlayContainer} />
         <Row>
-          {vertical ? (
-            <Col sm={12}>{renderSelectedWithDelete}</Col>
-          ) : (
-            <>
-              <Col sm={2} />
-              <Col sm={7}>{renderSelectedWithDelete}</Col>
-              <Col sm={3}>{renderExtraCol ? this.renderShortcuts() : null}</Col>
-            </>
-          )}
+          <Col sm={12}>{renderSelectedWithDelete}</Col>
         </Row>
       </>
     )
   }
 
   handleInputFocus = () => {
-    if (this.state.showOverlay === true) {
-      // Overlay is already open and we do input focus, no need to fetch data
-      this.setState({
-        inputFocused: true
-      })
-    } else {
-      this.setState(
-        {
-          searchTerms: "",
-          inputFocused: true,
-          showOverlay: true,
-          isLoading: true
-        },
-        this.fetchResults()
-      )
+    if (this.state.showOverlay) {
+      return // Overlay is already open and we do not need to fetch data
     }
+    this.setState(
+      {
+        searchTerms: "",
+        isLoading: true
+      },
+      this.fetchResults()
+    )
```

Comment:
This changes the behaviour: when clicking the icon (input field addon), which sets focus, the advanced select pop-up is no longer shown, only the input field gets focus.